account/recovery.go

```diff
@@ -220,16 +379,121 @@ func recoverInitialState(ctx context.Context, cfg RecoveryConfig) ([]*Account,
 func updateAccountStates(cfg RecoveryConfig, client *rpcclient.Client,
 	accounts []*Account) ([]*Account, error) {
 
-	// TODO (positiveblue): update account states
+	for i := cfg.FirstBlock; i < uint32(cfg.CurrentBlockHeight); i++ {
+		blockHeight := i
+
+		blockHash, err := client.GetBlockHash(int64(i))
+		if err != nil {
+			return nil, err
+		}
+
+		block, err := client.GetBlock(blockHash)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, tx := range block.Transactions {
+			for idx, acc := range accounts {
+				// It's safe to ignore accounts that are in a
+				// terminal state.
+				if acc.State == StateClosed {
+					continue
+				}
+				found := poolscript.IncludesPreviousOutPoint(
```

Comment:
Very nice, this now enables users to recover even if they needed to restore `lnd` from seed. And checking for the outpoint here should speed things up as much as possible.